qi-lib/flow/core/compiler/1000-qi0.rkt

```diff
@@ -167,15 +166,17 @@ already handled during expansion by Syntax Spec.
 
   (define (sep-parser stx)
     (syntax-parse stx
-      [_:id
-       #'(qi0->racket (if (esc list?)
-                          (#%fine-template (apply values _))
-                          (#%fine-template (raise-argument-error '△
-                                                                 "list?"
-                                                                 _))))]
-      [(_ onex:clause)
-       #'(λ (v . vs)
-           ((qi0->racket (~> △ (>< (#%fine-template (apply (qi0->racket onex) _ vs))))) v))]))
+      [(_ op:clause)
+       #'(zip-with (qi0->racket op))]
+      [_:id #'(λ args
+                (if (singleton? args)
+                    (let ([v (first args)])
+                      (if (list? v)
+                          (apply values v)  ; fast path to the basic △ behavior
+                          (raise-argument-error '△
+                                                "list?"
+                                                v)))
+                    (apply (qi0->racket (△ _)) args)))]))
```

Comment:
Maybe I'm missing it, but this seems like a change from the original `:id` clause that isn't mentioned by the commit or docs; perhaps it intended to be a separate kind of change "optimize common (sep)"?

Reply:
The original code expected a single list input, so I think this change was just to accommodate potentially more than one input. But I may have ended up optimizing it a bit by writing it fully in Racket? Not sure. The benchmark for `sep` did show a slight improvement I think.

Reply:
Oh, right. I guess I was looking at the "fast path" bit more than the overall computation.